demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -0,0 +1,561 @@
+                const_params.draw_ptr->DrawFPS(proc, 1e3f / (wait_time_ms / static_cast<float>(++wait_num_frames) + 1e-6f),
+                                               CV_RGB(0, 255, 0));
+            }
+
+            if (actions_type == TOP_K && monitoring_enabled) {
+                auto elapsed = std::chrono::high_resolution_clock::now() - started;
+                work_time_ms += std::chrono::duration_cast<std::chrono::milliseconds>(elapsed).count();
+                top_k.copyTo(const_params.draw_ptr->top_persons_);
+                const_params.draw_ptr->Show(const_params.draw_ptr->top_persons_, true);
+                const_params.draw_ptr->DrawFPS(proc, 1e3f / (work_time_ms / static_cast<float>(++work_num_frames) + 1e-6f),
+                                               CV_RGB(255, 0, 0));
+
+            } else if (actions_type != TOP_K) {
+                auto elapsed = std::chrono::high_resolution_clock::now() - started;
+                work_time_ms += std::chrono::duration_cast<std::chrono::milliseconds>(elapsed).count();
+                if (!vid_writer.isOpened()) {
+                    const_params.draw_ptr->DrawFPS(proc, 1e3f / (work_time_ms / static_cast<float>(work_num_frames) + 1e-6f),
+                                                   CV_RGB(255, 0, 0));
+                }
+                const_params.draw_ptr->Show(proc);
+                total_num_frames = work_num_frames;
+            }
+
+            if (vid_writer.isOpened()) {
+                vid_writer << proc;
+            }
+            if (FLAGS_last_frame >= 0 &&  work_num_frames > static_cast<size_t>(FLAGS_last_frame)) {
+                break;
+            }
+            std::cout << stream_log;
+        }
+        if (vid_writer.isOpened()) {
+            vid_writer.release();
+        };
+        const_params.draw_ptr->Finalize();
+        std::ofstream act_stat_log_stream, act_det_log_stream;
+        if (!FLAGS_al.empty()) {
+                act_det_log_stream.open(FLAGS_al,  std::fstream::out);
+                act_det_log_stream << "data" << "[" << std::endl;
+                act_det_log_stream << det_log << "]";
+        }
+        act_stat_log_stream.open(FLAGS_ad, std::fstream::out);
+        act_stat_log_stream << stat_log << std::endl;
+        slog::info << slog::endl;
+        if ( work_num_frames > 0) {
+            const float mean_time_ms = work_time_ms / static_cast<float>(work_num_frames);
+            slog::info << "Mean FPS: " << 1e3f / mean_time_ms << slog::endl;
+        }
+        slog::info << "Frames processed: " << total_num_frames << slog::endl;
+        std::cout << presenter.reportMeans() << '\n';
+    }
+    catch (const std::exception& error) {
+        slog::err << error.what() << slog::endl;
+        return 1;
+    }
+    catch (...) {
+        slog::err << "Unknown/internal exception happened." << slog::endl;
+        return 1;
+    }
+    slog::info << "Execution successful" << slog::endl;
+    return 0;
+}
```

Comment:
Please decompose `main()` into pieces. What I'd expect to see there is only building `GComputation` object. Everything else (e.g. preparing compile arguments) should be moved out.